packages/design-system/src/components/WIP/Tabs/Primitive/Tab.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { forwardRef, HTMLAttributes, Ref } from 'react';
+import { Tab as ReakitTab, TabState } from 'reakit';
+import classnames from 'classnames';
+// eslint-disable-next-line @talend/import-depth
+import { IconNameWithSize } from '@talend/icons/dist/typeUtils';
+import { DataAttributes } from '../../../../types';
+import { StackHorizontal } from '../../../Stack';
+import { SizedIcon } from '../../../Icon';
+
+import styles from './TabStyles.module.scss';
+import Tooltip from '../../../Tooltip';
+import { TagDefault } from '../../../Tag';
+
+type TabChildren =
+	| {
+			title: string;
+			icon?: IconNameWithSize<'S'>;
+			tag?: string | number;
+	  }
+	| { children: string };
+
+type TabSize = {
+	size?: 'M' | 'L';
+};
+
+type TabTooltip = {
+	tooltip?: string;
+};
+
+export type TabPropsTypesWithoutState = DataAttributes &
+	HTMLAttributes<HTMLButtonElement> &
+	TabSize &
+	TabTooltip &
+	TabChildren;
+
+type TabPropsTypes = TabPropsTypesWithoutState & TabState;
+
+const Tab = forwardRef((props: TabPropsTypes, ref: Ref<HTMLButtonElement>) => {
+	const { tooltip, ...otherProps } = props;
+
+	const component = () => {
```

Comment:
Do you know if there's an impact on performance between `component()` vs `<Component />` (the 2nd solution not dynamically declaring a component-like) ?
Just wondering, not world-changing point